doc/man7/openssl-threads.pod

```diff
@@ -0,0 +1,100 @@
+=pod
+
+=head1 NAME
+
+openssl-threads - Overview of thread safety in OpenSSL
+
+=head1 DESCRIPTION
+
+In this man page, we use the term B<thread-safe> to indicate that an
+object can be used by multiple threads at the same time.
+
+OpenSSL can be built with or without threads support. The most important
+use of this support is so that OpenSSL itself can use a single consistent
+API, as shown in L<CRYPTO_THREAD_run_once(3)/EXAMPLES>.
+Multi-platform applications can also use this API.
+
+In particular, being configured for threads support does not imply that
+all OpenSSL objects are thread-safe.
+To repeat: I<most objects are not safe for simultaneous use>.
+Exceptions to this should be documented on the specific manual pages, and
+some general high-level guidance is given here.
+
+One major use of the OpenSSL thread API is to implement reference counting.
+Many objects within OpenSSL are reference-counted, so resources are not
+released, until the last reference is removed.
+References are often increased automatically (such as when an B<X509>
+certificate object is added into an B<X509_STORE> trust store).
+There is often an I<object>_up_ref() API that can be used to increase the
+reference.
+Failure to match I<object>_up_ref() calls with the right number of
+I<object>_free() calls is a common source of memory leaks when a program
+exits.
+
+The various C<set1> and C<get1> functions associated with an object are
+used to set or get attributes of an object; they increase the reference
+count of the attribute, if the attribute is an object that has them.
+The C<set0> and C<get0> functions are similar, but do not increase the
+reference count, passing "ownership" appropriately.
+Functions with a simple C<set> or C<get> are historical; and the documentation
+must state how the references are handled.
+Get methods are often thread-safe as long as the ownership requirements are
+met.
+Set methods are generally not thread-safe as discussed below.
+
+Objects are thread-safe
+as long as the API's being invoked don't modify the object; in this
+case the parameter is usually marked in the API as C<const>.
+Not all parameters are marked this way.
+Note that a C<const> declaration does not mean immutable; for example
+L<X509_cmp(3)> takes pointers to C<const> objects, but the implementation
+uses a C cast to remove that so it can lock objects, generate and cache
+a DER encoding,, and so on.
```

Comment:
double comma here

Reply:
I don't think that this last sentence adds much value and just confuses the matter. The example given  is really just an implementation detail that should perhaps be noted in the man page for the function. IIUC, it doesn't impact the thread safety in any way.

Reply:
I think for the less-experienced C programmers this is useful info.